File: developer-guide/source/tutorials/webapps/common/resources/assets/standard.py
import dataiku
import pandas as pd
from flask import request

# Definition of the various resources (This can be done programmatically)
# ## Name of the managed folder
folder = "Resources"
# ## Path name of the files
image = "/image.png"
pdf = "/document.pdf"
css = "/my-css.css"


def get_file_from_managed_folder(folder_name, file_name):
    """
    Retrieve the file from the managed folder
    Args:
        folder_name: name of the managed folder
        file_name: name of the file

    Returns:
        the file
    """
    folder = dataiku.Folder(folder_name)

    with folder.get_download_stream(file_name) as f:
        return (f.read())

@app.route('/get_image_from_managed_folder')
def get_image_from_managed_folder():
    return get_file_from_managed_folder(folder, image)

@app.route('/get_pdf_from_managed_folder')
def get_pdf_from_managed_folder():
    return get_file_from_managed_folder(folder, pdf)

@app.route('/get_css_from_managed_folder')
def get_css_from_managed_folder():
    return get_file_from_managed_folder(folder, css)

File: developer-guide/source/tutorials/webapps/common/resources/assets/standard.js
let getImageFromManageFolderA    = document.getElementById('getImageFromManageFolderA')
let getImageFromManageFolderImg  = document.getElementById('getImageFromManageFolderImg')
let getImageFromManageFolderAImg = document.getElementById('getImageFromManageFolderAImg')
let getPDFFromManagedFolder      = document.getElementById('getPDFFromManagedFolder')
let getPDFFromManagedFolderA     = document.getElementById('getPDFFromManagedFolderA')
let getCSSFromManagedFolder      = document.getElementById('getCSSFromManagedFolder')

let getImageFromProjectLibImg    = document.getElementById('getImageFromProjectLibImg')
let getImageFromProjectLibA      = document.getElementById('getImageFromProjectLibA')
let getImageFromProjectLibAImg   = document.getElementById('getImageFromProjectLibAImg')
let getPDFFromProjectLibA        = document.getElementById('getPDFFromProjectLibA')
let getCSSFromProjectLib         = document.getElementById('getCSSFromProjectLib')

// Dynamic setting of the various element
getCSSFromManagedFolder.href     = getWebAppBackendUrl('/get_css_from_managed_folder')
getImageFromManageFolderA.href   = getWebAppBackendUrl('/get_image_from_managed_folder')
getImageFromManageFolderImg.src  = getWebAppBackendUrl('/get_image_from_managed_folder')
getImageFromManageFolderAImg.src = getWebAppBackendUrl('/get_image_from_managed_folder')
getPDFFromManagedFolderA.href    = getWebAppBackendUrl('/get_pdf_from_managed_folder')

getImageFromProjectLibImg.src  = `/local/projects/${dataiku.defaultProjectKey}/resources/image.jpg`
getImageFromProjectLibA.href   = `/local/projects/${dataiku.defaultProjectKey}/resources/image.jpg`
getImageFromProjectLibAImg.src = `/local/projects/${dataiku.defaultProjectKey}/resources/image.jpg`
getPDFFromProjectLibA.href     = `/local/projects/${dataiku.defaultProjectKey}/resources/file.pdf`
getCSSFromProjectLib.href      = `/local/projects/${dataiku.defaultProjectKey}/resources/my-css.css`


//
getPDFFromManagedFolder.addEventListener('click', () => {
    getWebAppBackendUrl('/get_pdf_from_managed_folder')
});

/* Javascript function for external URL */
let externalImageJs = document.getElementById('externalImageJs')

fetch("https://picsum.photos/200/300")
    .then(response => response.blob())
    .then(image => externalImageJs.src = URL.createObjectURL(image))